core/src/main/scala/org/apache/spark/sql/sources/SnappyOptimizations.scala

```diff
@@ -408,7 +409,13 @@ case class ResolveIndex(implicit val snappySession: SnappySession) extends Rule[
 
   override def apply(plan: LogicalPlan): LogicalPlan = {
     val joinOrderHints = JoinOrderStrategy.getJoinOrderHints
-
+    val enabled: Boolean = io.snappydata.Property.EnableExperimentalFeatures.
+        configEntry.getConf[Boolean](snappySession.snappyContext.conf) ||
+        java.lang.Boolean.getBoolean(io.snappydata.Property.EnableExperimentalFeatures
+            .configEntry.key)
```

Comment:
Ok, I pushed changes in 12119c1. So you can create this with alternative spark suffix. Then use:

```scala
EnableExperimentalFeatures.get[Boolean](snappySession.snappyContext.conf)
```

Spark prefixed system properties (and also "snappydata." ones in embedded mode) will automatically be picked up by SparkConf and from there will be copied to SQLConf by default so separate system property handling is not required and will work in split mode too.

Not required specifically for this I suppose since this is a temporary property, but this can be the general way for snappydata properties.